### PR TITLE
Issues/#1387 jsonldoptions

### PR DESCRIPTION
**Pull request by barthanssens**
GitHub issue resolved: #1387  

Briefly describe the changes proposed in this PR:

* Removed duplicate code in JSONLDParser
* Added ClassRioSettings
* Added PRODUCE_GENERALIZED_RDF and DOCUMENT_LOADER options
* Pass DOCUMENT_LOADER to JSONLDParser
* Pass COMPACT_ARRAYS and PRODUCE_GENERALIZED_RDF to JSONLDWriter
* Added test

### CODE REVIEW

**Comment by barthanssens** [comment]
@ansell , could you perhaps also take a look at this PR ? thanks

**Comment by barthanssens** [comment]
I covered the original use case / request on https://groups.google.com/forum/#!msg/rdf4j-users/xrirAvWzcP8/_GT91RicAwAJ + some minor enhancements

Though not entirely happy with the new ClassRioSetting

File: core/rio/api/pom.xml [hunk 1]
```diff
@@ -33,6 +33,11 @@
 			<version>${jaxb.version}</version>
 		</dependency>
 
+		<dependency>
+			<groupId>com.github.jsonld-java</groupId>
+			<artifactId>jsonld-java</artifactId>
+		</dependency>
```
Comment (jeenbroekstra) [inline core/rio/api/pom.xml]: I understand why you need this here and I'm ok with it, but it does pinpoint to me that in a future release we really should move the format-specific helpers out of the api module, and into their respective parser implementation modules.

Comment (barthanssens) [reply core/rio/api/pom.xml]: Indeed, probably a todo for 4.0 ...

Comment (barthanssens) [reply core/rio/api/pom.xml]: Created #1755

**Comment by barthanssens** [comment]
The ClassRioSetting felt like stretching it a bit, but it does indeed do the job...

**Comment by ansell** [comment]
I agree, having to hard-link a class (DocumentLoader) into the api module
is a sign that the format specific configurations are probably more suited
to being moved in the next major release.

The original rio config design had them all comfortably in the api module
because they were simple types that didn't hardlink into specific
implementations.

**Pull request merged**